network-firewall-logs.tf

```diff
@@ -0,0 +1,39 @@
+locals {
+  enable_firewall_logs = var.create_vpc && var.enable_firewall && var.enable_firewall_logs
+}
+
+########################
+# Network Firewall Logs
+########################
+resource "aws_networkfirewall_logging_configuration" "this" {
+  count = local.enable_firewall_logs && length(var.firewall_log_types) > 0 ? 1 : 0
+
```

Comment:
I get a transient error running this:

```
Error: 2 errors occurred:
        * error adding Logging Configuration to NetworkFirewall Firewall (arn:aws:network-firewall:us-east-1:123456789123:firewall/shared2-wallets-api-eks-vpc-firewall): ResourceNotFoundException: LogDestination: /aws/network-firewall-log/vpc-******************-flow does not exist
        * error adding Logging Configuration to NetworkFirewall Firewall (arn:aws:network-firewall:us-east-1:123456789123:firewall/shared2-wallets-api-eks-vpc-firewall): InvalidRequestException: Given logging configuration attempts to create/modify multiple log destination configs



  on vpc/network-firewall-logs.tf line 8, in resource "aws_networkfirewall_logging_configuration" "this":
   8: resource "aws_networkfirewall_logging_configuration" "this" {


Releasing state lock. This may take a few moments...
```

It looks like the `aws_networkfirewall_logging_configuration.this` resource needs a `depends_on` for `aws_cloudwatch_log_group.firewall_log`

Reply:
Have included this, thank you